doc/version-control-fields.rst

```diff
@@ -0,0 +1,179 @@
+Version Control System Fields
+=============================
+
+Most of the version control system (VCS) fields types are common to both
+``source-repository`` and ``source-repository-package`` stanzas.
+
+.. list-table::
+    :header-rows: 1
+    :widths: 30 30 40
+
+    * - Field Name
+      - source-repository (head|this)
+      - source-repository-package
+    * - type
+      - 🗹
+      - 🗹
+    * - location
+      - 🗹
+      - 🗹
+    * - branch
+      - 🗹
+      - 🗹
+    * - tag
+      - 🗹
+      - 🗹
+    * - subdir
+      - 🗹 (0 or 1)
+      - 🗹 (0 or 1 for each dependency)
+    * - module (CVS only)
+      - 🗹
+      - ☐
+    * - post-checkout-command
+      - ☐
+      - 🗹
+
+.. _vcs-kind:
+
+VCS kind
+^^^^^^^^
+
+Cabal supports specifying different information for various common source
+control systems. This is the name of the source control system used for a
+repository. The currently recognised types are:
+
+-  ``darcs``
+-  ``git``
+-  ``svn``
+-  ``cvs``
+-  ``mercurial`` (or alias ``hg``)
+-  ``bazaar`` (or alias ``bzr``)
+-  ``arch``
+-  ``monotone``
+-  ``pijul``
+
+The VCS kind will determine what other fields are appropriate to specify for a
+particular version control system.
+
+VCS location
+^^^^^^^^^^^^
+
+The location of the repository, usually a URL but the exact form of this field
+depends on the repository type. For example:
+
+-  for darcs: ``http://code.haskell.org/foo/``
+-  for git: ``git://github.com/foo/bar.git``
+-  for CVS: ``anoncvs@cvs.foo.org:/cvs``
+
+VCS branch
+^^^^^^^^^^
+
+Many source control systems support the notion of a branch, as a distinct
+concept from having repositories in separate locations. For example CVS, SVN and
+git use branches while darcs uses different locations for different branches. If
+you need to specify a branch to identify a your repository then specify it in
+this field.
+
+VCS tag
+^^^^^^^
+
+A tag identifies a particular state of a source repository.  The exact form of
+the tag depends on the repository type.
+
+VCS subdirectory
+^^^^^^^^^^^^^^^^
+
+A field of this type is always optional because it defaults to empty, which
+corresponds to the root directory of the repository and is the same as
+specifying ``.`` explicitly.
```

Comment:
These explanations paragraphs better belong to the field descriptions below. 

Also i'm still not convinced it makes sense to create a dedicated reference page to factor out the commonalities as the two use cases are rather independent (when i add a source repository stanza to add upstream metadata, i don't want to immediately add a VCS dependency, and vice versa). I would rather go for a bit of text duplication in the field descriptions to be able to read each stanza description on its own.
@ulysses4ever @andreabedini What do you think?

Reply:
@malteneuss how do you feel about this now, now that redirects to it describe their fields?

Reply:
It's better, yes, and good enough that i won't block this.